pkg/aws_keyhub/config.go

```diff
@@ -0,0 +1,119 @@
+package aws_keyhub
+
+import (
+	"encoding/json"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+var awsKeyHubConfigFile KeyhubConfigFile
+var doOnceReadAwsKeyHubConfig sync.Once
+
+func ConfigureAwsKeyhub() {
+	logrus.Println("aws-keyhub configuration wizard, please provide the following the information:")
+	var questions = []*survey.Question{
+		{
+			Name:     "keyHubUrl",
+			Prompt:   &survey.Input{Message: "KeyHub url (e.g. https://keyhub.domain.tld)"},
+			Validate: survey.Required,
+		},
+		{
+			Name:     "keyHubClientId",
+			Prompt:   &survey.Input{Message: "KeyHub aws-keyhub client id (e.g. 00000000-0000-0000-0000-000000000000"},
+			Validate: survey.Required,
+		},
+		{
+			Name:     "keyHubAwsSamlClientId",
+			Prompt:   &survey.Input{Message: "KeyHub Resource URN for the AWS SAML connection (e.g. urn:tkh-clientid:urn:amazon:webservices)"},
+			Validate: survey.Required,
+		},
+		{
+			Name:     "assumeDuration",
+			Prompt:   &survey.Input{Message: "AWS assume role duration (in seconds, maximum value is 43200) ", Default: "43200"},
+			Validate: survey.Required,
+		},
+	}
+	answers := struct {
+		KeyHubUrl             string
+		KeyHubClientId        string
+		KeyHubAwsSamlClientId string
+		AssumeDuration        int64
+	}{}
+
+	err := survey.Ask(questions, &answers)
+	if err != nil {
+		logrus.Fatal("Failed to prompt user for configuration settings.", err)
+	}
+
+	config := KeyhubConfigFile{
+		Aws: KeyhubAwsConfig{
+			AssumeDuration: answers.AssumeDuration,
+		},
+		Keyhub: KeyhubConfig{
+			Url:             answers.KeyHubUrl,
+			ClientId:        answers.KeyHubClientId,
+			AwsSamlClientId: answers.KeyHubAwsSamlClientId,
+		},
+	}
+
+	logrus.Debugln(config)
+	writeConfig(config)
+}
+
+type KeyhubConfigFile struct {
+	Keyhub KeyhubConfig    `json:"keyhub"`
+	Aws    KeyhubAwsConfig `json:"aws"`
+}
+
+type KeyhubConfig struct {
+	Url              string `json:"url"`
+	ClientId         string `json:"clientId"`
+	AwsSamlClientId  string `json:"awsSamlClientId"`
+	AllowInsecureTLS bool   `json:"allowInsecureTLS"` // We do not prompt for this flag, but it is configurable for development purposes.
+}
+
+type KeyhubAwsConfig struct {
+	AssumeDuration int64 `json:"assumeDuration"`
+}
+
+func CheckIfAwsKeyHubConfigFileExists() {
+	if _, err := os.Stat(getAwsKeyHubConfigFilePath()); os.IsNotExist(err) {
+		logrus.Fatal("It looks like you have no aws-keyhub configuration file. Please run `aws-keyhub configure` first.")
+	}
+	logrus.Debugln("aws-keyhub configuration file exists.")
+}
+
+func getAwsKeyHubConfigFilePath() string {
+	return getUserHomeDir() + string(os.PathSeparator) + ".aws-keyhub" + string(os.PathSeparator) + "config-v2.json"
+}
+
+func getAwsKeyHubConfig() KeyhubConfigFile {
+	doOnceReadAwsKeyHubConfig.Do(func() {
+		dat, err := ioutil.ReadFile(getAwsKeyHubConfigFilePath())
+		if err != nil {
```

Comment:
nil, dat is een weer een nieuwe variant van null of None denk ik? Hoe verzinnen ze het..